### DISCUSSION TITLE
"Renovate" for automated dependency updates

### DISCUSSION THREAD

**Discussion by elcajon** [homeassistant-apps/app-cloudflared#349] [Ideas]
I noticed that the [community add-on repository](https://github.com/hassio-addons/addon-base/pull/218) has recently started using [renovate](https://docs.renovatebot.com) for dependency updates. I have already removed the CI update logic in my add-on repositories and switched to renovate. The functionalities are pretty good and so far everything works without problems.
The big advantage is the function of automatic merging (PR) of configurable dependencies (e.g. only minor and patch versions may be merged automatically).
For this add-on, we could automate Docker base image as well as Cloudflared updates.

If there is interest, the [Renovate bot](https://github.com/apps/renovate) would need to be added to this repository. Afterwards an automatic PR will be generated in which I could set the configuration for the add-on.

**Comment by brenner-tobias** [comment]
Totally agree and great suggestion. I will have a look and create all the necessary thinks. Let's use this discussion as a information exchange until we are done with the transition.

**Comment by brenner-tobias** [comment]
I just added the renovate bot @elcajon

**Reply by elcajon** [reply]
I'll copy over my configuration and close this discussion

**Comment by brenner-tobias** [comment]
One thing I just saw @elcajon: We do have the dependency "ghcr.io/hassio-addons/base/amd64 13.2.0" from cloudflared/Dockerfile twice in the detected dependencies: once from dockerfile and once from regex.

It this a problem or is this fine? Can this be changed easily?

**Reply by elcajon** [reply]
This shouldn't be a problem. The bot will only create one PR (as it already did for the latest PR). The regex makes sure to combine both changes to one PR.
